src/model/Ticket.java

```diff
@@ -1,87 +1,110 @@
 package model;
 
 import enums.SectorType;
+import util.IdGenerator;
 
 import java.math.BigDecimal;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 import java.util.Currency;
 import java.util.Locale;
+import java.util.Objects;
 
-public class Ticket {
-    private String id;
-    private String concertHall;
-    private String eventCode;
+public class Ticket implements Printable,Identifiable {
+    private int classId;
+    private final String ticketId;
+    private final String concertHall;
+    private final String eventCode;
     private LocalDateTime time;
-    private boolean isPromo;
+    private final boolean isPromo;
     private SectorType stadiumSector;
-    private double maxAllowedBackpackWeightInKg;
+    private final double maxAllowedBackpackWeightInKg;
     private final LocalDateTime creationTime = LocalDateTime.now();
-    private BigDecimal ticketPrice;
+    private final BigDecimal ticketPrice;
 
     private static final String DATA_TIME_FORMAT = "yyyy.MM.dd, HH:mm";
 
     public Ticket() {
+        this.classId = IdGenerator.generateId();
+        this.ticketId = "";
+        this.concertHall = "";
+        this.eventCode = "";
+        this.isPromo = false;
+        this.maxAllowedBackpackWeightInKg = 0.0;
+        this.ticketPrice = BigDecimal.ZERO;
     }
 
-    public Ticket(String id, String concertHall,
+    public Ticket(String ticketId, String concertHall,
                   String eventCode,
                   LocalDateTime time,
                   boolean isPromo,
                   SectorType stadiumSector,
                   double maxAllowedBackpackWeightInKg,
                   BigDecimal ticketPrice) {
-        setId(id);
-        setConcertHall(concertHall);
-        setEventCode(eventCode);
+        this.classId = IdGenerator.generateId();
+        if (ticketId.length() > 4) {
+            throw new IllegalArgumentException("Size over 4 symbols");
+        }
+        if (concertHall.length() > 10) {
+            throw new IllegalArgumentException("Size over 10 chars");
+        }
+
+        if (eventCode.length() != 3 || !eventCode.matches("\\d{3}")) {
+            throw new IllegalArgumentException("Must be 3 digits");
+        }
+
+        if (maxAllowedBackpackWeightInKg < 0) {
+            throw new IllegalArgumentException("Invalid argument");
+        }
+
+        if (ticketPrice.compareTo(BigDecimal.ZERO) < 0) {
+            throw new IllegalArgumentException("Invalid argument");
+        }
+
+        this.ticketId = ticketId;
+        this.concertHall = concertHall;
+        this.eventCode = eventCode;
         this.time = time;
         this.isPromo = isPromo;
         setStadiumSector(stadiumSector);
```

Comment:
if there are no additional logic in  setStadiumSector method, than you can just assign the value:
this.stadiumSector = stadiumSector;

Reply:
done